test/kb_fasttree_server_test.py

```diff
@@ -37,92 +93,79 @@ def setUpClass(cls):
             cls.cfg[nameval[0]] = nameval[1]
-        cls.wsURL = cls.cfg["workspace-url"]
-        cls.wsClient = workspaceService(cls.wsURL, token=token)
-        cls.serviceImpl = kb_fasttree(cls.cfg)
+        cls.ws_client = Workspace(cls.cfg["workspace-url"], token=token)
+        cls.service_impl = kb_fasttree(cls.cfg)
+        # to regenerate the server responses, uncomment this line
+        # and delete the `test/cassettes` folder
+        # cls.run_id = datetime.now(tz=timezone.utc).strftime("%Y%m%d_%H%M%S")
+        # corresponds with the data in the saved cassette and the expected output files
+        cls.run_id = "20240801_204739"
 
     @classmethod
     def tearDownClass(cls):
-        if hasattr(cls, "wsName"):
-            cls.wsClient.delete_workspace({"workspace": cls.wsName})
-            print("Test workspace was deleted")
-
-    def getWsClient(self):
-        return self.__class__.wsClient
-
-    def getWsName(self):
-        if hasattr(self.__class__, "wsName"):
-            return self.__class__.wsName
-        suffix = int(time.time() * 1000)
-        wsName = "test_kb_fasttree_" + str(suffix)
-        ret = self.getWsClient().create_workspace({"workspace": wsName})
-        self.__class__.wsName = wsName
-        return wsName
-
-    def getImpl(self):
-        return self.__class__.serviceImpl
-
-    def getContext(self):
-        return self.__class__.ctx
-
-    ##############
-    # UNIT TESTS #
-    ##############
-
-    #### basic fasttree test
-    ##
-    # HIDE @unittest.skip("skipped test_kb_fasttree_run_FastTree_01()")  # uncomment to skip
-    def test_kb_fasttree_run_FastTree_01(self):
-        # Prepare test objects in workspace if needed using
-        # self.getWsClient().save_objects({'workspace': self.getWsName(), 'objects': []})
-        #
-        # Run your method by
-        # ret = self.getImpl().your_method(self.getContext(), parameters...)
-        #
-        # Check returned data with
-        # self.assertEqual(ret[...], ...) or other unittest methods
-
-        obj_basename = "fasttree"
-        obj_out_name = obj_basename + "." + "test_fasttree.Tree"
-        obj_out_type = "KBaseTrees.Tree"
-
-        # MSA
+        if hasattr(cls, "ws_name"):
+            try:
+                cls.ws_client.delete_workspace({"workspace": cls.ws_name})
+            except:
+                print("Workspace already deleted")
+            else:
+                print("Test workspace was deleted")
+
+    def get_ws_client(self):
+        return self.ws_client
+
+    def get_ws_name(self):
+        if hasattr(self, "ws_name"):
+            return self.ws_name
+        ws_name = f"test_kb_fasttree_{self.run_id}"
+        self.get_ws_client().create_workspace({"workspace": ws_name})
+        self.ws_name = ws_name
+        return ws_name
+
+    def get_impl(self):
+        return self.service_impl
+
+    def get_context(self):
+        return self.ctx
+
+    def get_msa_ref(self):
```

Comment:
This was part of the test itself but I moved it to be a separate function to make the test a bit clearer and to allow more tests to be run using the same `msa_ref`